src/Analyzer.JsonRuleEngine.UnitTests/AnyOfExpressionDefinitionTests.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.Templates.Analyzer.RuleEngines.JsonEngine.Expressions;
+using Microsoft.Azure.Templates.Analyzer.RuleEngines.JsonEngine.Operators;
+using Microsoft.Azure.Templates.Analyzer.RuleEngines.JsonEngine.Schemas;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace Microsoft.Azure.Templates.Analyzer.RuleEngines.JsonEngine.UnitTests
+{
+    [TestClass]
+    public class AnyOfExpressionDefinitionTests
+    {
+        [DataTestMethod]
+        [DataRow(1, DisplayName = "1 expression defined in AnyOf")]
+        [DataRow(5, DisplayName = "5 expressions defined in AnyOf")]
+        public void ToExpression_ValidExpressions_ReturnsArrayOfExpectedExpressions(int numberOfExpressionDefinitions, bool hasNullExpression = false)
```

Comment:
No tests use `hasNullExpression`.  I also think a `null` expression should be invalid, and should be blocked when parsing.

Reply:
When removing a data test row, I forgot to remove this parameter as well. Currently when a token is null, we just return null. This is in ReadJson in ExpressionConverter. Line 42. You're suggesting changing this to throw an error?

Reply:
See suggestion in other comment.